pkg/customers/customer_search.go

```diff
@@ -225,15 +225,18 @@ func (r *sqlCustomerRepository) getPhones(customerIDs []string) (map[string][]cl
 	for rows.Next() {
 		var p client.Phone
 		var customerID string
+		var phoneType PhoneType
+
 		err := rows.Scan(
 			&customerID,
 			&p.Number,
 			&p.Valid,
-			&p.Type,
+			&phoneType,
 		)
 		if err != nil {
 			return nil, fmt.Errorf("scanning row: %v", err)
 		}
+		p.Type = phoneType.Common()
```

Comment:
Maybe we can just create Valuer/Scanner for PhoneType and AddressType? It will remove the need in Common, etc. methods.

Reply:
When we generate client code we remove all files from /pkg/client. But we will need to keep validators, scanners/valuers, etc. Maybe it's also a good time to make something for this?

One way is to add prefix/posstfix to generated files like model_phone_gen.go? Then we can remove only generated files *_gen.go.

Reply:
Could you give me a small example of how that would work?

Reply:
Here is an example (no error handling in scanner / valuer):

In DB I created table:

```sql
create table addresses(id text, type int);
```
Here is code that inserts and loads values:

```go

type AddressType string

const (
	AddressTypePrimary   AddressType = "primary"
	AddressTypeSecondary AddressType = "secondary"
)

func (a *AddressType) Scan(value interface{}) error {
	intType := value.(int64)

	switch intType {
	case 1:
		*a = AddressTypePrimary
	case 2:
		*a = AddressTypeSecondary
	}

	return nil
}

func (a AddressType) Value() (driver.Value, error) {
	switch a {
	case AddressTypePrimary:
		return int64(1), nil
	case AddressTypeSecondary:
		return int64(2), nil
	}

	return nil, fmt.Errorf("unknown type")
}
```

having scanner and valuers allows us to do this:

```go
	stmt, err := db.Prepare("INSERT INTO addresses(id, type) VALUES( ?, ? )")
	if err != nil {
		log.Fatal(err)
	}
	defer stmt.Close() // Prepared statements take up server resources and should be closed after use.

        // here we just insert value of our custom type without converting it into integer
	if _, err := stmt.Exec("1", AddressTypeSecondary); err != nil {
		log.Fatal(err)
	}

       // later we can load it back from DB and get the same value
	row := db.QueryRow("select id, type from addresses where id='1'")
	var id int
        // we scan into AddressType but underlying value in db is int/int64
	var addressType AddressType
	err = row.Scan(&id, &addressType)
        
        if addressType == AddressTypeSecondary {
		// do whatever is needed
	}


```

Reply:
Working code: https://gist.github.com/alovak/b043f80c004dc895f5325ba63159dfa9

Reply:
Ah I see, this is useful, I'll add this since I won't have to convert it manually in my code in the data-access-layer.

But I think I still need methods `ToClient() ( formerly Common() )` and `addressTypeToModel(v client.AddressType) (AddressType, error)`; but I'm pretty sure I can use these methods in the `Scan/Value` methods.

I'll play around with it and update my PR.